language/move-prover/tests/sources/functional/macro_verification.move

```diff
@@ -0,0 +1,100 @@
+// flag: --trace
+
+// CVC5 does currently not terminate on this:
+// exclude_for: cvc5
+
+/// This file contains some simulated expansions of functional macros and sketches how they can be verified.
+module 0x42::FunMacros {
+    use Std::Vector;
+
+    /// Simulates `foreach!(v, |x| *x = *x + 1 spec  { ensures x == old(x) + 1; })`
```

Comment:
It is really nice that many of the harder loop invariants (related to framing, in particular) are hidden by the macro implementation.